src/production.f90

```diff
@@ -0,0 +1,306 @@
+    allocate(dvdy(0:nz, 0:ny-1, 0:nx-1))
+    allocate(dwdy(0:nz, 0:ny-1, 0:nx-1))
+
+    call field_default
+
+    if (l_vertical_profile) then
+        !
+        ! Height profile of production rates at a specific time
+        ! (averaged over x and y)
+        !
+        call open_netcdf_file(ncfname, NF90_NOWRITE, ncid)
+        call get_time_at_step(ncid, step, t)
+        call close_netcdf_file(ncid)
+
+        print *, 'Calculate enstrophy production rates at time', t
+
+        ! read field
+        call read_netcdf_fields(ncfname, step)
+
+        fname = ncfname(1:len(trim(ncfname))-3) // '_enstrophy_production_rates_step_' // trim(s_step) // '.asc'
+        open(unit=1235, file=fname, status='replace')
+        write(1235, *) '  # time', t
+        write(1235, *) '  # height eta_1 eta_2 eta_3'
+
+        do i = 0, nz
+            ! calculate the strain components
+            call calc_strain
+
+            ! calculate enstrophy production rates at specific height
+            ! (averaged over x and y)
+            call calc_production_rate_at_height(i)
+
+            ! write to file
+            z = lower(3) + dble(i) * dx(3)
+            write(1235, *) z, etas
+        enddo
+        close(1235)
+    else
+        !
+        !   Time evolution of individual production rates
+        !
+
+        call open_netcdf_file(ncfname, NF90_NOWRITE, ncid)
+        call get_num_steps(ncid, n_steps)
+        call close_netcdf_file(ncid)
+
+        ! 3 is subtracted to remove '.nc'
+        fname = ncfname(1:len(trim(ncfname))-3) // '_enstrophy_production_rates.asc'
+        open(unit=1235, file=fname, status='replace')
+        write(1235, *) '  # time (s) eta_1 eta_2 eta_3'
+        do step = 1, n_steps
+
+            ! read field
+            call read_netcdf_fields(ncfname, step)
+
+            call open_netcdf_file(ncfname, NF90_NOWRITE, ncid)
+            call get_time_at_step(ncid, step, t)
+            call close_netcdf_file(ncid)
+
+            ! calculate the strain components
+            call calc_strain
+
+            ! calculate enstrophy production rates
+            call calc_producition_rates
+
+            ! write to file
+            write(1235, *) t, etas
+        enddo
+        close(1235)
+    endif
+
+    deallocate(dudx)
+    deallocate(dudy)
+    deallocate(dwdx)
+    deallocate(dvdy)
+    deallocate(dwdy)
+
+    contains
+
+        subroutine calc_production_rate_at_height(iz)
+            integer, intent(in) :: iz
+            integer             :: ix, iy, nc
+            double precision    :: V(3, 3), D(3)
+
+            etas(:) = zero
+
+            do ix = 0, nx-1
+                do iy = 0, ny-1
+                    ! calculate eigenvalues and vectors of symmetrised strain matrix (1/2 * (S + S^T)
+                    call calc_eigs(ix, iy, iz, D, V)
+
+                    do nc = 1, 3
+                        etas(nc) = etas(nc) + D(nc) * sum(vor(iz, iy, ix, :) * V(:, nc)) ** 2
```

Comment:
Are you certain about the order of V(:, nc)?  Could it be V(nc, :)?  Eigenvectors can be stored strangely in Fortran

Reply:
Also, etas must be declared somewhere globally; I assume it is declared etas(3).

Reply:
Yes, I checked the order of the eigenvectors. All looks good.